### PR TITLE
Increase support for OpenCV_ARCH/OpenCV_RUNTIME flags

### PR DESCRIPTION
**Pull request by CSBVision**
Currently, running CMake Install on Windows copies the LIB and DLL files into `${OpenCV_ARCH}\${OpenCV_RUNTIME}\bin` and `${OpenCV_ARCH}\${OpenCV_RUNTIME}\lib` folders. Here, OpenCV_ARCH defaults to the respective architecture (e.g. x64) and OpenCV_RUNTME is usually defined by VS (e.g. vc17).

We would like to avoid these unnecessary directories (just as 'Install' copies the files on Linux as well as in other CMake projects), thus simply unset these variables. However, using empty variables results in CMake script errors during installation, while using "." works but uses paths with redundant `.\.\` parts. The proposal simply avoids the creation of these and does not prefixes the directories at the installation path at all if they are not set. If both variables are set to something else, nothing changes.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by alalek — COMMENTED** [review]
> avoid these unnecessary directories

Did you try `OPENCV_SKIP_CMAKE_ROOT_CONFIG=ON` ?

File: cmake/OpenCVInstallLayout.cmake [hunk 1]
```diff
@@ -22,8 +22,15 @@ if(ANDROID)
...
-  if(DEFINED OpenCV_RUNTIME AND DEFINED OpenCV_ARCH)
-    ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "${OpenCV_ARCH}/${OpenCV_RUNTIME}/")
+  if(DEFINED OpenCV_ARCH AND DEFINED OpenCV_RUNTIME)
+    if (NOT("${OpenCV_ARCH}"    STREQUAL "") AND NOT("${OpenCV_ARCH}"    STREQUAL "."))
+        ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "${OpenCV_ARCH}/")
+    else()
+        ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "")
+    endif()
+    if (NOT("${OpenCV_RUNTIME}" STREQUAL "") AND NOT("${OpenCV_RUNTIME}" STREQUAL "."))
+        set(OPENCV_INSTALL_BINARIES_PREFIX "${OPENCV_INSTALL_BINARIES_PREFIX}${OpenCV_RUNTIME}/")
```
[the remark this answers is not in the document]
Comment (CSBVision) [reply cmake/OpenCVInstallLayout.cmake:32]: Thanks for pointing out, we also tried `OPENCV_SKIP_CMAKE_ROOT_CONFIG=ON`, but still the runtime / arch directories are created, i.e. the respective code inside `OpenCVInstallLayout.cmake` will still be executed (you can also verify this as the code does not depend on this variable).
Regarding `ocv_update` and `set`: Yes correct, we also noted this. Still, sticking with `ocv_update` requires to build the prefix path structure in an additional variable (e.g. `prefix`) and after it is created (using `set`), only use `ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "${prefix}")` once.
We tried using `ocv_update(OPENCV_INSTALL_BINARIES_PREFIX "${OPENCV_INSTALL_BINARIES_PREFIX}${OpenCV_RUNTIME}/")`, still this does not set the path correctly as the runtime part will be missing (but we think this is the expected behavior of `ocv_update`).
If you prefer using this prefix variable, we are fine with that and update the PR.

**Comment by CSBVision** [comment]
After your comments, we investigated this folder structure creation issue once again. It seems that explicitly defining `OPENCV_INSTALL_BINARIES_PREFIX` as empty avoids these folders as well. Still, the CMake scripts will define `OpenCV_ARCH` and `OpenCV_RUNTIME` according to the build configuration (i.e. usually `x64` and `vcXX`, depending on the current Visual Studio version) but the respective directories are not created by running CMake's install target.

We don't know whether it is advantageous to define `OPENCV_INSTALL_BINARIES_PREFIX` instead and ignoring the other two flags. Besides being at least inconsistent, the result seems to be fine as these two flags have no other purpose. What do you think about this solution, is directly defining `OPENCV_INSTALL_BINARIES_PREFIX` and ignoring the flags fine? Than the PR might not be necessary anymore. This configuration seems to be rather internal and not directly documented somewhere, so at least this information might be helpful for anybody running into the same issue.

**Comment by alalek** [comment]
You probably want layout something like `openvino_opencv_windows.tgz` distribution from https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.1/opencv/
See `build_params.txt` (inside of the archive) for used CMake options.

P.S. Ensure to cleanup CMake cache (or build directory) between experiments.

**Comment by CSBVision** [comment]
Yes, that's pretty much what we want to achieve using these flags. If explicitly defining the directories as in your linked `build_params.txt` is fine for the CMake setup, we think that this is even the more elegant solution such that the PR is no longer required. If we saw this before, we wouldn't have created this PR. Thanks 👍

**Pull request closed without merging**